roles/letsencrypt/tasks/dns-challenge-azure.yml

```diff
@@ -0,0 +1,88 @@
+- name: create TXT record in a new record set
+  azure_rm_dnsrecordset:
+    resource_group: "{{ azure_resource_group }}"
+    relative_name: "_acme-challenge"
+    zone_name: "{{ domain.zone }}"
+    record_mode: append
+    state: present
+    record_type: TXT
+    records:
+      - entry: "{{ challenge['challenge_data'][domain.common_name]['dns-01']['resource_value'] }}"
+
+# split top_level for zone_name and if subdomain is defined add subdomain to relative_name
+- name: add a new TXT record to the SAN top-level domains
+  azure_rm_dnsrecordset:
+    resource_group: "{{ azure_resource_group }}"
+    relative_name: "{{ '_acme-challenge' if item == item.split('.')[-2:] | join('.') else '_acme-challenge.' + item.split('.')[:-2] | join('.') }}"
```

Comment:
Since this is fairly hard to read code, can you please add an example (here: https://github.com/T-Systems-MMS/ansible-collection-letsencrypt/tree/master/examples)?

Reply:
add examples